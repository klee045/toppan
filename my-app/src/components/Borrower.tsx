import React, { CSSProperties } from "react";

const BorrowerStyle: CSSProperties = {
  marginTop: "20px",
  padding: "30px",
  borderRadius: "10px",
  boxShadow: "3px 6px 5px gray",
  backgroundColor: "rgb(255, 215, 215)",
  fontSize: "24px",
};

const Borrower = ({ name }: { name: string }): JSX.Element => {
  return (
    <div className="customer" style={BorrowerStyle}>
      {name}
    </div>
  );
};

export default Borrower;
