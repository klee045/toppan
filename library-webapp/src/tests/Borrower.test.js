import React from "react";
import renderer from "react-test-renderer";
import Borrower from "../components/Borrower";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

it("renders correctly", () => {
  const tree = renderer.create(<Borrower name={"Jon Hamm"} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('has class "customer"', () => {
  const tree = renderer.create(<Borrower name={"Jon Hamm"} />);
  expect(
    shallow(<Borrower name={"Jon Hamm"} />)
      .find("div.customer")
      .exists()
  ).toBe(true);
});
