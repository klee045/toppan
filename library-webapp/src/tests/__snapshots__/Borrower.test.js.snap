// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="customer"
  style={
    Object {
      "backgroundColor": "rgb(255, 215, 215)",
      "borderRadius": "10px",
      "boxShadow": "3px 6px 5px gray",
      "fontSize": "24px",
      "marginTop": "20px",
      "padding": "30px",
    }
  }
>
  Jon Hamm
</div>
`;
